src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -1448,6 +1492,118 @@ private function step_in_body() {
+		$tag_name = $this->get_tag();
+		$op_sigil = $this->is_tag_closer() ? '-' : '+';
+		$op       = "{$op_sigil}{$tag_name}";
+
+		switch ( $op ) {
+			/*
+			 * > A start tag whose tag name is "html"
+			 */
+			case '+HTML':
+				$this->last_error = self::ERROR_UNSUPPORTED;
+				throw new WP_HTML_Unsupported_Exception( "Cannot process {$tag_name} element." );
+
+			/*
+			 * > A start tag whose tag name is "option"
+			 */
+			case '+OPTION':
+				$current_node = $this->state->stack_of_open_elements->current_node();
+				if ( $current_node && 'OPTION' === $current_node->node_name ) {
+					$this->state->stack_of_open_elements->pop();
+				}
+				$this->insert_html_element( $this->state->current_token );
+				return true;
+
+			/*
+			 * > A start tag whose tag name is "optgroup"
+			 * > A start tag whose tag name is "hr"
+			 */
+			case '+OPTGROUP':
+			case '+HR':
+				$current_node = $this->state->stack_of_open_elements->current_node();
+				if ( $current_node && 'OPTION' === $current_node->node_name ) {
+					$this->state->stack_of_open_elements->pop();
+					// If we've popped, update the current_node
+					$current_node = $this->state->stack_of_open_elements->current_node();
+				}
+
+				if ( $current_node && 'OPTGROUP' === $current_node->node_name ) {
+					$this->state->stack_of_open_elements->pop();
+				}
+
+				$this->insert_html_element( $this->state->current_token );
+				return true;
+
+			/*
+			 * > An end tag whose tag name is "optgroup"
+			 */
+			case '-OPTGROUP':
+				$walker       = $this->state->stack_of_open_elements->walk_up();
+				$current_node = $walker->current();
+				if ( ! $current_node ) {
+					return $this->step();
+				}
+				if ( 'OPTGROUP' === $current_node->node_name ) {
+					$this->state->stack_of_open_elements->pop();
+					return true;
+				}
+
+				$walker->next();
+				$current_node_parent = $walker->current();
+				if ( 'OPTION' === $current_node->node_name && 'OPTGROUP' === $current_node_parent->node_name ) {
+					$this->state->stack_of_open_elements->pop();
+					$this->state->stack_of_open_elements->pop();
+					return true;
+				}
+				return $this->step();
+
+			/*
+			 * > An end tag whose tag name is "option"
+			 */
+			case '-OPTION':
+				$current_node = $this->state->stack_of_open_elements->current_node();
+				if ( $current_node && 'OPTION' === $current_node->node_name ) {
+					$this->state->stack_of_open_elements->pop();
+					return true;
+				}
+				return $this->step();
+
+			/*
+			 * > An end tag whose tag name is "select"
+			 * > A start tag whose tag name is "select"
+			 */
+			case '-SELECT':
+			case '+SELECT':
+				if ( ! $this->state->stack_of_open_elements->has_element_in_select_scope( 'SELECT' ) ) {
+					return $this->step();
+				}
+				$this->state->stack_of_open_elements->pop_until( 'SELECT' );
+				$this->state->stack_of_open_elements->pop();
+				$this->reset_insertion_mode();
+				return true;
+		}
+
+		$this->last_error = self::ERROR_UNSUPPORTED;
```

Comment:
noting that INPUT, KEYGEN, and TEXTAREA are missing

Reply:
and SCRIPT and TEMPLATE

Reply:
Done.